pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -341,6 +347,64 @@ func (h *NetworkUtilsHandler) GenerateRandomMac() (net.HardwareAddr, error) {
 	return net.HardwareAddr(append(prefix, suffix...)), nil
 }
 
+func (h *NetworkUtilsHandler) CreateTapDevice(tapName string, isMultiqueue bool, launcherPID string) error {
+	launcherSELinuxLabel, err := selinux.FileLabel(fmt.Sprintf("/proc/%s/attr/current", launcherPID))
+	if err != nil {
+		log.Log.Reason(err).Errorf("Could not retrieve the file label associated w/ PID: %s", launcherPID)
+		return err
+	}
+
+	synchNamespaces := []string{
+		"--mount", "/proc/1/ns/mnt",
```

Comment:
Could you explain why the switch to the node mount-namespace is needed and not for instance a switch to the launcher namespace? How is the tap device then finally accessible inside the launcher namespace?

Reply:
> `/proc/<handler-pid>/task/<current-tid>/attr/exec`.

Did you mean `launcher-pid` her?

Could you explain that in a little bit more detail?

Reply:
Switching to the node's mount ns **is** needed to prevent the following AVC (which again exhibits the behavior of unprivileged MCS virt-handler accessing unprivileged MCS virt-launcher):

```
type=AVC msg=audit(1596123813.304:2894): avc:  denied  { read } for  pid=25002 comm="tap-device-make" path="/dev/null" dev="tmpfs" ino=1538376 sco
ntext=system_u:system_r:virt_launcher.process:s0:c167,c346 tcontext=system_u:object_r:container_file_t:s0:c971,c983 tclass=chr_file permissive=0

```

@rmohr on our talk, I've mentioned this behavior **but** for the tap creation binary itself. I hope this 'counts' as proof that virt-handler (when running with the launcher's context - featuring MCS) cannot access stuff with MCS from the handler's container.

Reply:
:+1 yes that is good enough for me :)